components/script/textinput.rs

```diff
@@ -276,7 +276,7 @@ impl<T: ClipboardProvider> TextInput<T> {
         }
 
         debug_assert!(self.edit_point.line < self.lines.len());
-        debug_assert!(self.edit_point.index <= self.lines[self.edit_point.line].len());
+        debug_assert!(self.edit_point.index <= max(1, self.lines[self.edit_point.line].len()));
```

Comment:
I need to take a look at what's going on on the test-case exactly, but I suspect this is wrong.

In particular, this should be `self.lines[self.edit_point.line].len() + 1` instead, but in any case, I still _suspect_ it's the wrong solution, and the `edit_point` should end up being `1, 0` instead of `0, 1` in this case, since we have two lines in the test-case.